### PR TITLE
Maker codes admin ui

### PR DESCRIPTION
**Pull request by epeach**
Admin Override UI for discount maker codes.

### CODE REVIEW

File: apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
+/** @file Admin Override for Maker Discount Code Eligibility*/
+import React, {Component, PropTypes} from 'react';
+import i18n from "@cdo/locale";
+import SchoolAutocompleteDropdown from '../../../../templates/SchoolAutocompleteDropdown';
+import Button from "../../../../templates/Button";
+import ValidationStep, {Status} from '../../../../lib/ui/ValidationStep';
```
Comment (Bjvanminnen) [inline apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx]: I personally really don't like seeing these paths with lots of `../` in them. I would prefer to see an absolute reference instead, i.e. `from '@cdo/apps/lib/ui/validationStep';`.

That said, there's some argument in favor in the relative paths in that I think some peoples' editors know how to follow them, but not our absolute paths.

File: apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    errors: {
+      invalidEmail: false
+    }
+  };
+
+  handleDropdownChange = (field, event) => {
+    this.setState({
+      submission: {
+        ...this.state.submission,
+        [field]: event ? event.value : ''
```
Comment (Bjvanminnen) [inline apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx]: when is this called without an event?

File: apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        <h1 style={styles.title}>Circuit Playground Kits Admin Override</h1>
+        <form>
+          <label>
+            <div>Teacher email address or username (for the account they are using in the classroom):</div>
+            <input
+              value=""
+              ref={input => this.teacherID = input}
+            />
+            <Button
+              color="orange"
```
Comment (Bjvanminnen) [inline apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx]: Button.ButtonColor.orange

File: apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+              setField={this.handleDropdownChange}
+              value={submission.nces}
+              showErrorMsg={errors.nces}
+            />
+            <h2>Admin Options</h2>
+            <h4>Option 1: Link teacher account with other accounts</h4>
+            <div>
+              If teacher meets the eligibity requirements but is simply using a
+               different email address for their account than what we have on file,
+               please go to the Studio Person ID admin page to link this acccount
```
Comment (Bjvanminnen) [inline apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx]: I think spec calls for there to be a link to the page in here?

File: apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+               to the saccount associated with the email address we have on file.
+            </div>
+            <h4>Option 2: Give teacher a discount code</h4>
+            <div>
+              If the teacher has already received their code and it was for the incorrect amount,
+              email adaaccounts@adafruit.com so they can disable that code first. We should not be
```
Comment (Bjvanminnen) [inline apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx]: could probably also make this a mailto link

File: apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx (hunk 4 above)
Comment (epeach) [reply apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx]: Was waiting for the link. Just received and adding now.

File: apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx (hunk 1 above)
Comment (epeach) [reply apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx]: Yeah, I don't love the inconsistency on this. My editor can follow the "../" paths, but @ cdo is much cleaner. I think I might try to stick with @ cdo though.

File: apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    this.setState({
+      submission: {
+        ...this.state.submission,
+        [field]: event.value
+      }
+    });
+  }
+
+  handleSubmitId = () => {
+    this.setState({teacherID : this.teacherID.value});
```
Comment (Bjvanminnen) [inline apps/src/lib/kits/maker/ui/DiscountAdminOverride.jsx]: nit: no space before `:`

File: apps/src/lib/kits/maker/ui/DiscountAdminOverride.story.jsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import DiscountAdminOverride from './DiscountAdminOverride';
+import {Status} from '../../../../lib/ui/ValidationStep';
```
Comment (Bjvanminnen) [inline apps/src/lib/kits/maker/ui/DiscountAdminOverride.story.jsx]: another place you could use `cdo/apps` if you wanted

**Pull request merged**